doc/manual.md

```diff
@@ -7049,6 +7049,22 @@ immediate pragma
 The immediate pragma is obsolete. See `Typed vs untyped parameters
 <#templates-typed-vs-untyped-parameters>`_.
 
+redefine pragma
+---------------
+
+Redefinition of template symbols with the same signature is allowed.
+This can be made explicit with the `redefine` pragma:
+
+```nim
+template foo: int = 1
+echo foo() # 1
+template foo: int {.redefine.} = 2
+echo foo() # 2
+# warning: implicit redefinition of template
+template foo: int = 3
+```
+
+This is mostly intended for macro generated code. 
```

Comment:
````suggestion
While redefining template with the same signature is allowed, doing so without using the `redefine` pragma will cause the compiler to emit a warning:

```nim
template implicitFoo: int = 1
echo implicitFoo() # 1

template implicitFoo: int = 2 # warning: implicit redefinition of template
echo implicitFoo() # 2
```
```
template explicitFoo: int = 3
echo explicitFoo() # 3

template explicitFoo: int = 4 # no warning
echo explicitFoo() # 4
```
````

Reply:
There is no {.redefine.} on explicitFoo. Also the line about macros is removed, not sure if intended.

Reply:
Woops, you're right. Let me fix that.

Reply:
Regarding the "this is mostly intended for macros" - unless this warning only applies to redefinitions caused by macro-generated templates, I'd say it applies to any template redefinition.

Reply:
It does apply to any template redefinition, the feature just mostly makes sense with the context of macros. Even if it's not exclusive to macros I think it could at least be mentioned that it *can* be used in macros